frontend/js/src/common/listens/ListenCard.tsx

```diff
@@ -96,706 +94,615 @@ export type ListenCardProps = {
+    Boolean(recordingMBID) ||
+    artistMBIDs?.length ||
+    Boolean(trackMBID) ||
+    Boolean(releaseGroupMBID);
+
+  const hasActionOptions =
+    additionalMenuItems?.length ||
+    hasInfoAndMBID ||
+    recordingMBID ||
+    spotifyURL ||
+    youtubeURL ||
+    soundcloudURL;
+  const hideActionsMenu = compact || !hasActionOptions;
+
+  const renderBrainzplayer =
+    userPreferences?.brainzplayer?.brainzplayerEnabled ?? true;
+
+  let timeStampForDisplay;
+  if (customTimestamp) {
+    timeStampForDisplay = customTimestamp;
+  } else if (displayListen.playing_now) {
+    timeStampForDisplay = (
+      <span className="listen-time">
+        <Link to="/listening-now/">
+          <FontAwesomeIcon icon={faMusic as IconProp} /> Listening now &#8212;
+        </Link>
+      </span>
+    );
+  } else {
+    timeStampForDisplay = (
+      <span
+        className="listen-time"
+        title={
+          displayListen.listened_at
+            ? fullLocalizedDateFromTimestampOrISODate(
+                displayListen.listened_at * 1000
+              )
+            : fullLocalizedDateFromTimestampOrISODate(
+                displayListen.listened_at_iso
+              )
+        }
       >
-        <div className="main-content">
-          {beforeThumbnailContent}
-          {thumbnail}
-          {listenDetails ? (
-            <div className="listen-details">{listenDetails}</div>
-          ) : (
-            <div className="listen-details">
-              <div className="title-duration">
-                <div
-                  title={trackName ?? releaseName}
-                  className={compact ? "ellipsis" : "ellipsis-2-lines"}
-                >
-                  {trackName ? getTrackLink(listen) : getAlbumLink(listen)}
-                </div>
-                {trackDurationMs && (
-                  <div className="small text-muted" title="Duration">
-                    {isNumber(trackDurationMs) &&
-                      millisecondsToStr(trackDurationMs)}
-                  </div>
-                )}
-              </div>
-              <div className="small text-muted ellipsis" title={artistName}>
-                {getArtistLink(listen)}
+        {preciseTimestamp(
+          displayListen.listened_at_iso || displayListen.listened_at * 1000
+        )}
+      </span>
+    );
+  }
+
+  const thumbnail =
+    customThumbnail ??
+    getThumbnailElement(
+      thumbnailSrc,
+      displayListen,
+      currentUser,
+      openMBIDMappingModal
+    );
+
+  return (
+    <Card
+      {...otherProps}
+      onDoubleClick={renderBrainzplayer ? playListen : undefined}
+      className={`listen-card ${isCurrentlyPlaying ? "current-listen" : ""}${
+        compact ? " compact" : ""
+      }${additionalContent ? " has-additional-content" : " "} ${
+        className || ""
+      }`}
+      data-testid="listen"
+    >
+      <div className="main-content">
+        {beforeThumbnailContent}
+        {thumbnail}
+        {listenDetails || (
```

Comment:
You need to revert this change and add a parent div with class `listen-details`.

The top-album and top-track stats break because of this.

Reply:
<img width="306" height="613" alt="image" src="https://github.com/user-attachments/assets/df9a2f8d-7209-4a9a-b9c1-33b597ad55ef" />